filedescriptor/src/unix.rs

```diff
@@ -217,6 +266,33 @@ impl FileDescriptor {
         }
         Ok(())
     }
+
+    /// Attempt to duplicate the underlying handle from an object that is
+    /// representable as the system `RawFileDescriptor` type and assign it to
+    /// a destination file descriptor. It then returns a `FileDescriptor`
+    /// wrapped around the duplicate.  Since the duplication requires kernel
+    /// resources that may not be available, this is a potentially fallible operation.
+    /// The returned handle has a separate lifetime from the source, but
+    /// references the same object at the kernel level.
+    pub unsafe fn dup2<F: AsRawFileDescriptor>(f: &F, dest_fd: RawFd) -> anyhow::Result<Self> {
+        OwnedHandle::dup2_impl(f, dest_fd).map(|handle| Self { handle })
+    }
+
+    pub(crate) fn redirect_stdio_impl<F: AsRawFileDescriptor>(
+        f: &F,
+        stdio: StdioDescriptor,
+    ) -> anyhow::Result<Self> {
+        let std_descriptor = match stdio {
+            StdioDescriptor::Stdin => libc::STDIN_FILENO,
+            StdioDescriptor::Stdout => libc::STDOUT_FILENO,
+            StdioDescriptor::Stderr => libc::STDERR_FILENO,
+        };
+
+        let std_original = FileDescriptor::dup(std_descriptor)?;
+        unsafe { FileDescriptor::dup2(f, std_descriptor) }?;
```

Comment:
I think this may also need `.into_raw_fd()` here to avoid closing the fd when this function returns.

It would be great to add a regression test for this :)

Reply:
I've made the change to add ```.into_raw_fd()``` into the code. Not sure what do make for the regression test. I had a plan instead to implement this now in a branch in the gag crate. Was gonna run my tests for these through the gag create instead.